build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/Utils.java

```diff
@@ -934,21 +874,11 @@ static public int scan(StringBuffer buf, int start, String[] targets) {
 	 * @throws IOException
 	 */
 	static public StringBuffer readFile(File target) throws IOException {
-		return readFile(new FileInputStream(target));
+		return new StringBuffer(new String(Files.readAllBytes(target.toPath())));
 	}
 
 	static public StringBuffer readFile(InputStream stream) throws IOException {
-		StringBuffer result = new StringBuffer();
-		char[] buf = new char[4096];
-		int count;
-		try (InputStreamReader reader = new InputStreamReader(new BufferedInputStream(stream))) {
-			count = reader.read(buf, 0, buf.length);
-			while (count != -1) {
-				result.append(buf, 0, count);
-				count = reader.read(buf, 0, buf.length);
-			}
-		}
-		return result;
+		return new StringBuffer(new String(stream.readAllBytes()));
```

Comment:
I think this method is sufficiently simple to just inline it into the callers.
The callers of this method should also use StringBuilder instead of StringBuffer.

Reply:
That would duplicate the flaw that a std charset is used.